packages/core/src/amazonqFeatureDev/session/sessionState.ts

```diff
@@ -282,24 +288,40 @@ abstract class CodeGenBase {
                         codeGenerationTotalIterationCount: codeGenerationTotalIterationCount,
                     }
                 }
-                case 'predict-ready':
-                case 'InProgress': {
+                case CodeGenerationStatus.PREDICT_READY:
+                case CodeGenerationStatus.IN_PROGRESS: {
                     await new Promise((f) => globals.clock.setTimeout(f, this.requestDelay))
                     break
                 }
-                case 'predict-failed':
-                case 'debate-failed':
-                case 'Failed': {
-                    /** 
-                     * 
-                     * TODO: Here we need to implement the granular error handling for 
-                     *  Code generation GuardrailsException
-                        Code generation PromptRefusalException
-                        Code generation EmptyPatchException
-                        Code generation ThrottlingException
-                     * 
-                     * */
-                    throw new ToolkitError('Code generation failed', { code: 'CodeGenFailed' })
+                case CodeGenerationStatus.PREDICT_FAILED:
+                case CodeGenerationStatus.DEBATE_FAILED:
+                case CodeGenerationStatus.FAILED: {
+                    switch (true) {
+                        case codegenResult.codeGenerationStatusDetail?.includes('Guardrails'): {
+                            throw new FeatureDevServiceError(
+                                "I'm sorry, I'm having trouble generating your code. Please try again.",
```

Comment:
Has there been any thought on de-duping some of the strings? I'm seeing the same string on line 311

Reply:
That's a good one. We don't have in VSCode toolkit a .properties file like in JB, but we somehow can avoid these duplications. I'll let this for we dedupe in a specific task to handle i18n, if you're ok with it.

Reply:
`package.nls.json` is a good usecase for this, but even a simple top-level variable is better than duplication here.

Reply:
Sounds good, I'll push a rev removing this duplication. Thanks

Reply:
Done. @hayemaxi are we good to merge this one?